gradoop-common/src/main/java/org/gradoop/common/model/impl/pojo/Element.java

```diff
@@ -62,6 +62,7 @@ protected Element(
     this.id = id;
     this.label = label;
     this.properties = properties;
+    setProperty("__label__", label);
```

Comment:
labels must not be part of the properties. decision logic needs to be in predicate evaluation